app/src/main/java/com/github/onlynotesswent/ui/overview/FolderContent.kt

```diff
@@ -93,6 +94,17 @@ fun FolderContentScreen(
   var updatedName by remember { mutableStateOf(folder.value!!.name) }
   var expandedFolder by remember { mutableStateOf(false) }
 
+  // Custom back handler to manage back navigation
+  BackHandler {
```

Comment:
Great Idea to use a custom back handler as it creates a custom navigation experience.

Reply:
Yes, the intention is that using the back button when in `FolderContent` will only allow to navigate to the previous folder and not to edit note in case we came from a note